app/app_kas.c

```diff
@@ -456,8 +456,27 @@ int app_kas_ifc_handler(ACVP_TEST_CASE *test_case) {
 
     tc = test_case->tc.kas_ifc;
 
-    /** Step 1: Convert all existing values into bignum */
+    if (tc->md != ACVP_NO_SHA) {
+        printf("Hash alg provided for KAS-IFC case but not supported\n");
+        goto err;
+    }
+
+    /** Step 1: Convert all existing values into bignum, null check needed values */
     if (tc->kas_role == ACVP_KAS_IFC_INITIATOR || tc->scheme == ACVP_KAS_IFC_KAS2) {
+
+        if (!tc->server_n || !tc->server_e || !tc->server_nlen || !tc->server_elen) {
+            printf("Missing server pub key params in KAS-IFC test case\n");
+            goto err;
+        }
+        if (!tc->iut_pt_z || !tc->iut_ct_z) {
+            printf("Missing buffer for IUT Z in KAS-IFC test case\n");
+            goto err;
+        }
+        if (tc->test_type == ACVP_KAS_IFC_TT_VAL && !tc->iut_pt_z_len) {
+            printf("Needed provided IUT PT Z for KAS-IFC test case, but is not provided\n");
```

Comment:
"Need to provide IUT PT Z for KAS-IFC test case"

Reply:
Will update in a next PR, thanks